pallets/task/src/lib.rs

```diff
@@ -499,7 +563,8 @@ pub mod pallet {
 			Ok(())
 		}
 
-		pub fn delete_task(task_initiator: &T::AccountId, task_id: &T::Hash) -> Result<(), DispatchError> {
+		pub fn accept_completed_task(task_initiator: &T::AccountId, task_id: &T::Hash) -> Result<(), DispatchError> {
+			
 			// Check if task exists
 			let mut task = Self::tasks(&task_id).ok_or(<Error<T>>::TaskNotExist)?;
 
```

Comment:
is_task_initiator() checks for task exists or not. So perphas Self::tasks() can be removed and just directly mutate task after is_task_initiator() check.

Reply:
Good catch... However, we are using `let mut task`, to get the current task struct and do mutations on it.